superset/assets/src/explore/components/ExploreChartHeader.jsx

```diff
@@ -94,7 +94,6 @@ class ExploreChartHeader extends React.PureComponent {
       chartUpdateStartTime,
       latestQueryFormData,
       queryResponse } = this.props.chart;
-    const chartSucceeded = ['success', 'rendered'].indexOf(this.props.chart.chartStatus) > 0;
```

Comment:
this `success` state is very critical. for a chart, it has many states in its life cycle: start fetching, fetch stopped by user, fetch failed, fetch timeout, fetch succeeded, render succeed, render failed, etc. There are only 2 states means data is ready and you want to use updated data to re-render chart. otherwise, either you start render without good data, or repeated render same data again and again :(